### PR TITLE
Ignore mem_gb=0 in autoexec

### PR DESCRIPTION
**Pull request by jrapin**
Context: being able to **not** pass `mem` parameter so as to be able to use `mem-per-gpu` in the hydra plugin: https://github.com/facebookresearch/hydra/issues/1366

### CODE REVIEW

File: submitit/slurm/test_slurm.py [hunk 1]
```diff
@@ -345,7 +347,7 @@ def test_name() -> None:
 
 
 @contextlib.contextmanager
-def with_slurm_job_nodelist(node_list: str):
+def with_slurm_job_nodelist(node_list: str) -> tp.Generator[slurm.SlurmJobEnvironment, None, None]:
```
Comment (gwenzek) [inline submitit/slurm/test_slurm.py:350]: tp.Iterator[slurm.SlurmJobEnvironment]

**Comment by gwenzek** [comment]
I'm not convinced by the fix.
Why do we need to do this ? Why Hydra is always setting `mem` ? Shouldn't this be deferred to the user ?

I'd rather report a conflict between `mem` and `mem_per_gpu`, or just ignore `mem` when `mem_per_gpu` is set.

File: submitit/slurm/test_slurm.py (hunk 1 above)
Comment (jrapin) [inline submitit/slurm/test_slurm.py:350]:
```suggestion
def with_slurm_job_nodelist(node_list: str) -> tp.Iterator[slurm.SlurmJobEnvironment]:
```

**Comment by jrapin** [comment]
> I'm not convinced by the fix.
> Why do we need to do this ? Why Hydra is always setting `mem` ? Shouldn't this be deferred to the user ?

There are default values in the configuration, which sets a default value for mem. This is in the process of being removed anyway, so that one can set the value to `None`, which will not be passed to submitit in this case.

> I'd rather report a conflict between `mem` and `mem_per_gpu`, or just ignore `mem` when `mem_per_gpu` is set.

I'm not sure I do, out of consistency. If mem had not been a string but a int, it would have had the new behavior automatically because null (0) and None values are ignored. And I would not ignore `mem` if `mem_per_gpu` is set: what happens if you then set `mem_per_task` as well? which should have the priority? it becomes a much bigger mess

**Comment by jrapin** [comment]
@gwenzek see alternative #1607 which seems to fit better what hydra would need https://github.com/facebookresearch/hydra/pull/1375

**Comment by gwenzek** [comment]
I think I prefer #1608 as it feels less special cased

**Pull request closed without merging**